default.nix

```diff
@@ -1,17 +1,72 @@
-with import (builtins.fetchGit {
-  url = https://github.com/NixOS/nixpkgs-channels;
-  ref = "nixos-18.09-small";
-  rev = "95fed28ac372c61eb83c87ad97c24b0f957827bf";
-}) {};
-
-stdenv.mkDerivation {
-  name = "postgrest-docs";
-  buildInputs = [
-    python36Full
-    python36Packages.sphinx
-    python36Packages.sphinx_rtd_theme
-    python36Packages.livereload ];
-  shellHook = ''
-    python livereload_docs.py && exit
-  '';
+let
+  # Commit of the Nixpkgs repository that we want to use.
+  nixpkgsVersion = {
+      date = "2020-10-04";
+      rev = "d191ee22f0b0c91070f9f4c79210d4e398df0b93";
+      tarballHash = "13fb4cgxxc47shmzbm0rawfxjnilf7r4zq1vwc0z6s3lxijl9wl9";
+    };
+
+  # Nix files that describe the Nixpkgs repository. We evaluate the expression
+  # using `import` below.
+  nixpkgs =
+    builtins.fetchTarball {
+      url = "https://github.com/nixos/nixpkgs/archive/${nixpkgsVersion.rev}.tar.gz";
+      sha256 = nixpkgsVersion.tarballHash;
+    };
+in
+
+with import nixpkgs {};
+let
+  python = python36.withPackages (ps: [ps.sphinx ps.sphinx_rtd_theme ps.livereload]);
+
+  build =
+    writeShellScriptBin "postgrest-docs-build"
+      ''
+        set -euo pipefail
+
+        # clean previous build, otherwise some errors might be supressed
+        rm -rf _build
+
+        ${python}/bin/sphinx-build -W -b html -a -n . _build
+      '';
+
+  serve =
+    writeShellScriptBin "postgrest-docs-serve"
+      ''
+        set -euo pipefail
+
+        # livereload_docs.py needs to find "sphinx-build"
+        PATH=${python}/bin:$PATH
+
+        ${python}/bin/python livereload_docs.py
+      '';
+
+  spellcheck =
+    writeShellScriptBin "postgrest-docs-spellcheck"
+      ''
+        set -euo pipefail
+
+        cat *.rst                                                                              \
+         | grep -v '^\(\.\.\|  \)'                                                             \
+         | sed 's/`.*`//g'                                                                     \
+         | ${aspell}/bin/aspell -d ${aspellDicts.en}/lib/aspell/en_US -p ./postgrest.dict list \
+         | tee misspellings
+        test ! -s misspellings
+      '';
+in
+
+{
+  build = { inherit build; };
```

Comment:
I think `inherit build serve spellcheck;` instead of the three lined here should to the trick!

Reply:
Unfortunately no, then `nix-shell` will complain about too many derivations. I put all 3 in a subkey `tools` at first, but this would only allow me to install all 3 of them together in CI. That's why I split them up.

Reply:
The error is: `error: nix-shell requires a single derivation`

Reply:
But your sneaky suggestion from above works. and looks better.